tests/prague/eip7692_eof_v1/eip7620_eof_create/test_eofcreate_failures.py

```diff
@@ -585,3 +585,189 @@ def test_insufficient_returncontract_auxdata_gas(
+        + Op.STOP
+    )
+    calling_address = pre.deploy_contract(
+        Container.Code(calling_code) if opcode == Op.EXTSTATICCALL else calling_code
+    )
+
+    post = {
+        calling_address: Account(
+            storage={
+                slot_call_result: value_eof_call_result_failed
+                if opcode == Op.EXTSTATICCALL
+                else value_legacy_call_result_failed,
+                slot_code_worked: value_code_worked,
+            }
+        )
+    }
+    tx = Transaction(
+        to=calling_address,
+        gas_limit=10_000_000,
+        protected=False,
+        sender=sender,
+    )
+
+    state_test(env=env, pre=pre, post=post, tx=tx)
+
+
+magic_value_call = 0xCA11
+magic_value_create = 0xCC12EA7E
+
+
+@pytest.mark.parametrize(
+    "opcode",
+    [
+        Op.EXTCALL,
+        Op.EXTDELEGATECALL,
+    ],
+)
+@pytest.mark.parametrize(
+    "who_fails",
+    [magic_value_call, magic_value_create],
+    ids=["call_fails", "create_fails"],
+)
+def test_eof_eofcreate_msg_depth(
+    state_test: StateTestFiller,
+    pre: Alloc,
+    opcode: Op,
+    who_fails: int,
+):
+    """
+    Test EOFCREATE handles msg depth limit correctly (1024).
+    NOTE: due to block gas limit and the 63/64th rule this limit is unlikely to be hit
+          on mainnet.
+    NOTE: See `tests/prague/eip7692_eof_v1/eip7069_extcall/test_calls.py::test_eof_calls_msg_depth`
+          for more explanations and comments. Most notable deviation from that test is that here
+          calls and `EOFCREATE`s alternate in order to reach the max depth. `who_fails` decides
+          whether the failing depth 1024 will be on a call or on an `EOFCREATE` to happen.
+    """
+    # Not a precise gas_limit formula, but enough to exclude risk of gas causing the failure.
+    gas_limit = int(20000000 * (64 / 63) ** 1024)
+    env = Environment(gas_limit=gas_limit)
+    sender = pre.fund_eoa()
+
+    callee_address = Address(0x5000)
+
+    returndatacopy_block = Op.RETURNDATACOPY(64, 0, 96) + Op.REVERT(64, 96)
+    deep_most_result_block = (
+        Op.MSTORE(64, Op.ADD(Op.CALLDATALOAD(0), 1)) + Op.MSTORE(96, Op.NOOP) + Op.REVERT(64, 96)
+    )
+    rjump_offset = len(returndatacopy_block)
+
+    callee_code = Container(
+        sections=[
+            Section.Code(
+                Op.MSTORE(0, Op.ADD(Op.CALLDATALOAD(0), 1))
+                + Op.MSTORE(128, magic_value_create)
+                + Op.EOFCREATE[0](0, Op.CALLDATALOAD(0), 0, 64)
+                + Op.RETURNDATASIZE
+                + Op.ISZERO
+                + Op.RJUMPI[rjump_offset]
+                + returndatacopy_block
+                + deep_most_result_block
+            ),
+            Section.Container(
+                Container.Code(
+                    Op.MSTORE(0, Op.ADD(Op.CALLDATALOAD(0), 1))
+                    + Op.MSTORE(128, magic_value_call)
+                    + opcode(address=callee_address, args_size=64)
```

Comment:
This isn't so simple, as this is initcode, so `Op.ADDRESS` is different at everydepth, also such in-deployment contract cannot call itself... I guess, I'm a bit :exploding_head: with this... Maybe let's keep it simple here and just have a static `callee_address`?

Reply:
Yes I understand, let's just add the `@pytest.mark.pre_alloc_modify` marker to this test in order to not try to run it in live networks (it's impossible simply because of the amount of gas consumed anyway).

You might need to rebase to `main` to get the marker.